andela/ninjas-ah-backend
# #165412918  Give a rating to an article

#### What does this PR do?
Give a rating on the article written by others

#### Description of Task to be completed?
Have the following endpoint accessible to an authenticated user:

`POST /api/v1/articles/:slug/rating` give rating to an article identified by its slug
`GET /api/v1/articles/:slug/rating` Get users who rated a given article

#### How should this be manually tested?
- After cloning the **repo**, `cd` into it and type `npm test`
- Using Postman, 
->make sure you signup then login to get the access token, 
->create an article,
-> Create a rating using this endoint: `/api/v1/articles/:slug/rating`
-> Make sure you pass the access token in the header with the following key name: `access-token`,
-> Finally, send a **JSON** object of the following structure : {"rating":  1}

#### Any background context you want to provide?
- The formula used to calculate a rating of a given article
Given,
`n`:  number of rating per article
`s`: sum of ratings per a given article

```
rating: s/n
```

#### What are the relevant pivotal tracker stories?
[#165412918](https://www.pivotaltracker.com/story/show/165412918)

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A
